pandas/tests/indexing/test_iloc.py

```diff
@@ -1186,6 +1186,17 @@ def test_iloc_setitem_series_duplicate_columns(self):
         df.iloc[:, 0] = df.iloc[:, 0].astype(np.float64)
         assert df.dtypes.iloc[2] == np.int64
 
+    def test_iloc_setitem_dtypes_duplicate_columns(self):
+        # GH#22035
+        df = DataFrame([["0", "str", "1.2"]], columns=["a", "b", "b"])
```

Comment:
please paramterize over dtypes

Reply:
My apologies for missing the parametrization earlier